### PR TITLE
Fix indefinite retries if the store cannot be reached

### PR DESCRIPTION
**Pull request by lachypjones**
# Description
Simple, small quality of life fix for the current fake store implementation. This will time out a failed connection to the store after 10 seconds.

# Self-review quick check

 * [x] Make decisions that cost a lot to reverse explicit in the PR description.
 * [x] Avoid nested conditions.
 * [x] Delete dead code and redundant comments.
 * [x] Normalise symmetries by sticking to doing identical things identically. 
 ```
 // one way to handle errors
 if err := f(); err != nil {
    ...
 }
 
 // one way to handle multiple returns
 val, err := f()
 if err != nil {
    ...
 }
 ...
 ```
 * [x] Check that coupled code elements, files, and directories are adjacent. For example, test data is stored as close as possible to a test.
 * [x] Put variable declaration and initialisation together.
 * [x] Divide large expressions into digestable and self-explanatory ones. Use multiple variables if required.
 * [x] Put a blank line between two logically different chunks of code.

## Docs

* [ ] I have checked and added or updated relevant documentation.
* [ ] I have checked and added or updated relevant release notes.
* [ ] I have included the technical author in the review.

Or:

* [x] I confirm the PR has no implications for documentation.

### CODE REVIEW

File: internal/fakestore/store.go [hunk 1]
```diff
@@ -219,6 +219,10 @@ func storeSdkInfoImpl(ctx context.Context, name, channel string) (storeSdk, erro
 	}
 	track, risk := sa[0], sa[1]
 	obj := bkt.Object(fmt.Sprintf("%s/%s/%s/%s.sdk", name, track, risk, name))
+
+	// Attrs transparently retries forever if it cannot connect to the store
+	ctx, cancel := context.WithTimeout(ctx, time.Second*10)
```
Comment (dmitry-lyfar) [inline internal/fakestore/store.go]: @lachypjones we need a unit test for this, as per the comment, is it going to retry forever on every timeout?

**Comment by akcano** [comment]
The issue appears transient and unrelated to the latest commit:
image

File: internal/fakestore/store.go [hunk 2]
```diff
@@ -219,11 +220,21 @@ func storeSdkInfoImpl(ctx context.Context, name, channel string) (storeSdk, erro
...
+	// passed on to SDK download behavior
+	obj = obj.Retryer(
+		storage.WithMaxAttempts(3),
+	)
 	atr, err := obj.Attrs(ctx)
 	if err != nil {
 		if errors.Is(err, storage.ErrObjectNotExist) {
 			return sSdk, fmt.Errorf("SDK not found in %q", channel)
 		}
+		urlError := &url.Error{}
```
Comment (dmitry-lyfar) [inline internal/fakestore/store.go]: It's not enough to test for url.Error, we should return "failed to connect" on the network errors:
```
	if urlErr, ok := err.(*url.Error); ok {
	    opErr, ok := urlErr.Err.(*net.OpError)
            if ok {
		// cannot connect
	     }
	}
```

**Comment by lachypjones** [comment]
@dmitry-lyfar 
For the error message, do you prefer it with the network error included:
```
error: SDK store action failed:
- test: Cannot connect to store: "dial tcp 127.0.0.1:8080: connect: connection refused"
```
Or without:
```
error: SDK store action failed:
- test: Cannot connect to store
```

**Comment by dmitry-lyfar** [comment]
the network error would be great to include, yes. also, lower case.

**Review by dmitry-lyfar — APPROVED** [review]

**Pull request merged**